include/ppqsort/parallel/cpp/thread_pool.h

```diff
@@ -112,11 +111,11 @@ namespace ppqsort::impl::cpp {
                 total_tasks_.fetch_sub(1, std::memory_order_release);
             }
```

Comment:
As this is no longer a `jthread` which *j*oins automatically when the destructor is called, the `.join()` needs to be done manually here for every worker thread.